std/file.d

```diff
@@ -930,10 +931,13 @@ if (isConvertibleToString!R)
     assert(getSize(deleteme.byChar) == 3);
 }
 
-
-// Reads a time field from a stat_t with full precision.
-version(Posix)
-private SysTime statTimeToStdTime(char which)(ref stat_t statbuf)
+version(StdDdoc)
+{
+    /// Reads a time field from a stat_t with full precision.
+    SysTime statTimeToStdTime(char which)(const ref stat_t statbuf) if (which == 'a' || which == 'c' || which == 'm');
+}
+else version(Posix)
+SysTime statTimeToStdTime(char which)(const ref stat_t statbuf) if (which == 'a' || which == 'c' || which == 'm')
```

Comment:
As a rule of thumb we should use template parameters only if there's a good chance the user will have the arguments in the same format. In this case, it's unlikely someone has a compile-time char that is `a`, `m`, or `c` received from elsewhere so a template is not warranted here.